src/tests/all-tests.js

```diff
@@ -0,0 +1,138 @@
+require("../../node_modules/gpii-express/src/js/router");
+require("../../node_modules/gpii-express/src/js/middleware");
+require("../../node_modules/gpii-express/src/js/bodyparser");
+require("../js/pouch.js");
+
+function isSaneResponse(jqUnit, error, response, body, status) {
+    var status = status ? status : 200;
+    jqUnit.assertNull("There should be no errors.", error);
+
+    jqUnit.assertEquals("The response should have a reasonable status code", status, response.statusCode);
+    if (response.statusCode !== status) {
+        console.log(JSON.stringify(body, null, 2));
+    }
+
+    jqUnit.assertNotNull("There should be a body.", body);
+};
+
+var sampleDataFile = path.resolve(__dirname, "./data/data.json");
+var userDataFile   = path.resolve(__dirname, "./data/users.json");
+var pouch = gpii.express({
+    "config": {
+        "express": {
+            "port" :   7532,
+            "baseUrl": "http://localhost:7532/"
+        }
+    },
+    components: {
+        "pouch": {
+            type: "gpii.pouch",
+            options: {
+                model: {
+                    "databases": {
+                        "_users": { "data": userDataFile },
+                        "data":   { "data": sampleDataFile },
+                        "nodata": {}
+                    }
+                }
+            }
+        }
+    }
+});
+
+pouch.start(function() {
+    jqUnit.module("Testing pouch module stack...");
+
+    jqUnit.asyncTest("Testing the root of the pouch instance...", function() {
+        var options = {
+            url: pouch.options.config.express.baseUrl
+        };
+        request.get(options, function(error, response, body){
+            jqUnit.start();
+            isSaneResponse(jqUnit, error,response, body);
+        });
+    });
+
+    jqUnit.asyncTest("Testing the 'nodata' database (should not contain data)...", function() {
+        var options = {
+            url: pouch.options.config.express.baseUrl + "nodata"
+        };
+        request.get(options, function(error, response, body){
+            jqUnit.start();
+            isSaneResponse(jqUnit, error,response, body);
+
+            var data = (typeof body === "string") ? JSON.parse(body) : body;
+            jqUnit.assertEquals("There should be no records.", 0, data.doc_count);
+        });
+    });
+
+    jqUnit.asyncTest("Testing the 'data' database (should contain data)...", function() {
+        var options = {
+            url: pouch.options.config.express.baseUrl + "data"
+        };
+        request.get(options, function(error, response, body){
+            jqUnit.start();
+            isSaneResponse(jqUnit, error,response, body);
+
+            var data = (typeof body === "string") ? JSON.parse(body) : body;
+            jqUnit.assertTrue("There should be records.", data.doc_count && data.doc_count > 0);
+        });
+    });
+
+    // TODO:  test inserts
+    jqUnit.asyncTest("Testing insertion of a new record...", function() {
```

Comment:
As before with gpii-express, express this repetitive test material declaratively